actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -90,18 +101,18 @@ class ActionCable.Connection
     open: ->
       ActionCable.log("WebSocket onopen event")
       @disconnected = false
-      @subscriptions.reload()
+      if not @isProtocolSupported()
+        ActionCable.log("WebSocket got '#{@getProtocol()}' as the chosen protocol. Stopping monitor and disconnecting.")
+        @close(allowReconnect: false)
+        return
+      ActionCable.log("WebSocket using '#{@getProtocol()}' subprotocol.")
```

Comment:
Not loving how this method reads now. The early `return` only guards against logging. I suggest:
```coffee
open: ->
  ActionCable.log("WebSocket onopen event, using '#{@getProtocol()}' subprotocol")
  @disconnected = false
  unless @isProtocolSupported()
    ActionCable.log("Protocol is unsupported. Stopping monitor and disconnecting.")
    @close(allowReconnect: false)
```